mpu6050/mpu6050.go

```diff
@@ -1,95 +1,251 @@
-// Package mpu6050 provides a driver for the MPU6050 accelerometer and gyroscope
-// made by InvenSense.
-//
-// Datasheets:
-// https://store.invensense.com/datasheets/invensense/MPU-6050_DataSheet_V3%204.pdf
-// https://www.invensense.com/wp-content/uploads/2015/02/MPU-6000-Register-Map1.pdf
-package mpu6050 // import "tinygo.org/x/drivers/mpu6050"
+package mpu6050
+
+import (
+	"encoding/binary"
+	"fmt"
+
+	"tinygo.org/x/drivers"
+)
 
-import "tinygo.org/x/drivers"
+const DefaultAddress = 0x68
 
-// Device wraps an I2C connection to a MPU6050 device.
-type Device struct {
-	bus     drivers.I2C
-	Address uint16
+type IMUConfig struct {
+	// Use ACCEL_RANGE_2 through ACCEL_RANGE_16.
+	AccRange byte
+	// Use GYRO_RANGE_250 through GYRO_RANGE_2000
+	GyroRange   byte
+	sampleRatio byte // TODO(soypat): expose these as configurable.
+	clkSel      byte
 }
 
-// New creates a new MPU6050 connection. The I2C bus must already be
-// configured.
-//
-// This function only creates the Device object, it does not touch the device.
-func New(bus drivers.I2C) Device {
-	return Device{bus, Address}
+// Dev contains MPU board abstraction for usage
+type Dev struct {
+	conn    drivers.I2C
+	aRange  int32 //Gyroscope FSR acording to SetAccelRange input
+	gRange  int32 //Gyroscope FSR acording to SetGyroRange input
+	data    [14]byte
+	address byte
+}
+
+// New instantiates and initializes a MPU6050 struct without writing/reading
+// i2c bus. Typical I2C MPU6050 address is 0x68.
+func New(bus drivers.I2C, addr uint16) *Dev {
+	p := &Dev{}
+	p.address = uint8(addr)
+	p.conn = bus
+	return p
+}
+
+// Init configures the necessary registers for using the
+// MPU6050. It sets the range of both the accelerometer
+// and the gyroscope, the sample rate, the clock source
+// and wakes up the peripheral.
+func (p *Dev) Init(data IMUConfig) (err error) {
```

Comment:
This is called `Configure` on most drivers.

```suggestion
func (p *Dev) Configure(data IMUConfig) (err error) {
```